doc/deps.md

```diff
@@ -303,3 +303,5 @@ relevant during compilation.
 | `DATABASE` | Database type. Possible values are **sqlite**, **pgsql**. The default value is `sqlite`. |
 | `TEST_DB` | The connection string for the database that will be used when executing tests with `make test`. Optional. |
 | `CODECOMPASS_LINKER` | The path of the linker, if the system's default linker is to be overridden. |
+| `WITH_PLUGIN` | The names of the plugins to be built. Possible values are **cpp**, **cpp_reparse**, **dummy**, **git**, **metrics**, **search**. The metrics and search plugins are fundamental, they will be compiled even if not included. Keep in mind that the cpp_reparse plugin cannot be compiled without the cpp plugin. `WITH_PLUGIN` **cannot** be used together with `WITHOUT_PLUGIN`. Example: `-DWITH_PLUGIN="cpp;git"` This will compile the cpp, git, metrics and search plugins. |
+| `WITHOUT_PLUGIN` | The names of the plugins to be skipped at build. Possible values are **cpp**, **cpp_reparse**, **dummy**, **git**, **metrics**, **search**. The metrics and search plugins are fundamental, they will be compiled even if included. Keep in mind that the cpp_reparse plugin cannot be compiled without the cpp plugin. `WITHOUT_PLUGIN` **cannot** be used together with `WITH_PLUGIN`. Example: `-DWITHOUT_PLUGIN="dummy;git;search"` This will compile the cpp, cpp_reparse, metrics and search plugins. |
```

Comment:
It the option cannot be used for `metrics` and `search` (*), I think it is enough if a too eager user is told by the CMake error message. No need to mention in the "beginner's guide" something that shouldn't happen and we can also reasonably guard against anyways.

(*): Although, **why** can't it? I understand the reasoning behind not having the ability to disallow `search`, but why is `metrics` such a core thing? I hardly ever cared about the output of that plugin, ever...)

Reply:
Same with `cpp_reparse`: there's a warning or error if `cpp_reparse` is on and `cpp` isn't, or at least there *should* be. This superfluous explanation makes the documentation hard to follow.

Reply:
There is a warning message if someone tries to exclude them. However, if you find this description too much, I'll remove the explanations and merge these two lines into one line in the table.

We talked about it a couple weeks ago, and @zporky said that metrics could be considered a baseline plugin. I agreed because it's such a small and universal plugin, it won't hurt if we make it obligatory. I thought that we could include search, too.

Reply:
Alright, understandable. Not saying it hurts, just that it was surprising at first glance!